### ISSUE TITLE
Addition of Search Functionality to Ma'akaf Open-Source Community Website

### ISSUE THREAD

**Issue by eyallampel1** [Maakaf/maakaf-temp#55]
@UrielOfir I'd like to suggest adding a search functionality to the Ma'akaf Open-Source Community website. Having a search feature would improve the site’s usability, allowing users to effortlessly navigate and find the information or projects they are interested in.

Here’s a simple breakdown of what the search functionality might include:

Search Bar: An intuitive search bar where users can type in keywords.
Dynamic Results: As users type, relevant results will appear dynamically, improving the user experience.

I would like to take the initiative to implement it. Could this issue be assigned to me? I am open to discussing the design, functionality, and any other details to make this feature as effective and useful as possible.

**Comment by UrielOfir** [comment]
Sounds great.
I've assigned you this task.

**Comment by UrielOfir** [comment]
What search engine will know to do this search work?

**Comment by UrielOfir** [comment]
You can use my booking page to schedule a meeting- (
https://calendar.app.google/zt6SqPQrYP8PHZ468)[https://calendar.app.google/zt6SqPQrYP8PHZ468]
.

**Comment by eyallampel1** [comment]
@UrielOfir, I've checked your schedule and the earliest available slot is on Sunday. I've gone ahead and reserved a time early in the day before my work hours.

Meanwhile, please take a moment to review the functionality of the search feature as currently implemented on the site:

![Search Feature Demo](https://github.com/eyallampel1/assets/blob/main/maakaf_search.gif?raw=true)

**Comment by UrielOfir** [comment]
Your video of the search feature looks amazing.

**Comment by eyallampel1** [comment]
@UrielOfir,

Thank you for the positive feedback on the search feature demonstration. I'm pleased to share that I have continued to refine the functionality, taking into account both our discussions and several design iterations to ensure the best possible user experience.

To ensure accessibility and responsiveness, I have tested the new search feature across various screen sizes and devices. I'm happy to report that it presents well on all of them, providing a consistent and user-friendly experience irrespective of the user's device.

I have implemented a small magnifying glass icon, available in both English and Hebrew, that aligns with the website's design look. When a user clicks on this icon, they are smoothly transitioned to a dedicated search page, relevant to the language mode they are in. This dedicated search page allows for a focused environment to review search results without altering the overall aesthetic or navigational flow of the site.

An animated GIF is attached to this message to illustrate the feature in action. It shows the interaction from the point of clicking the magnifying glass icon to navigating through the search results. This solution, I believe, offers a good blend of design and functionality, enhancing the site's utility without disrupting its visual .

Furthermore, to facilitate the review process, I have initiated a pull request linked to this issue. This will allow for a thorough review of the code changes and the new feature in the context of the existing codebase.

![Search Feature Demo2](https://github.com/eyallampel1/assets/blob/main/maakaf_site_search_magnify.gif?raw=true)
